### PR TITLE
{devel}[GCCcore/12.3.0, system] Bazel v6.3.1, Java v11.0.20, Java v11

### PR DESCRIPTION
**Pull request by maxim-masterov**
New Bazel and Java11

### CODE REVIEW

**Comment by casparvl** [comment]
@boegelbot please test @ generoso

**Comment by casparvl** [comment]
Test report by @casparvl
**SUCCESS**
Build succeeded for 4 out of 4 (3 easyconfigs in total)
tcn52.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/0bf0fc9dedf660ece3d5bc5bceeba1ca for a full test report.

**Comment by casparvl** [comment]
@boegelbot please test @ jsc-zen2

File: easybuild/easyconfigs/b/Bazel/Bazel-6.3.1-GCCcore-12.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+homepage = 'https://bazel.io/'
+description = """Bazel is a build tool that builds code quickly and reliably.
+It is used to build the majority of Google's software."""
+
+toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
+
+source_urls = ['https://github.com/bazelbuild/%(namelower)s/releases/download/%(version)s']
+sources = ['%(namelower)s-%(version)s-dist.zip']
+# patches = [
```
[the remark this answers is not in the document]
Comment (maxim-masterov) [reply easybuild/easyconfigs/b/Bazel/Bazel-6.3.1-GCCcore-12.3.0.eb:12]: Both patches have already been incorporated into the source codes. Bazel-6.3.1 relies on protobuf-3.19.6, that has already added `use_default_shell_env = True` to `protobuf.bzl` [see here](https://github.com/protocolbuffers/protobuf/blob/12f743d4e7bc2a22918371a286e9ab09748b0f40/protobuf.bzl#L360)

**Review by casparvl — APPROVED** [review]
Lgtm!

**Comment by casparvl** [comment]
Going in, thanks @maxim-masterov!

**Pull request merged**